src/com/projectkorra/projectkorra/airbending/combo/AirSweep.java

```diff
@@ -191,18 +192,22 @@ public void manageAirVectors() {
 						this.remove();
 						return;
 					}
-					if (!entity.equals(this.player) && !this.affectedEntities.contains(entity) && !(entity instanceof Player && Commands.invincible.contains(((Player) entity).getName()))) {
-						this.affectedEntities.add(entity);
+					if (!entity.equals(this.player) && !(entity instanceof Player && Commands.invincible.contains(((Player) entity).getName()))) {
 						if (this.knockback != 0) {
-							final Vector force = fstream.getDirection();
-							entity.setVelocity(force.multiply(this.knockback));
+							final Vector force = fstream.getLocation().getDirection();
+							GeneralMethods.setVelocity(entity, force.clone().multiply(this.knockback));
+							new HorizontalVelocityTracker(entity, this.player, 200l, this);
+							entity.setFallDistance(0);
 						}
-						if (this.damage != 0) {
-							if (entity instanceof LivingEntity) {
-								if (fstream.getAbility().getName().equalsIgnoreCase("AirSweep")) {
-									DamageHandler.damageEntity(entity, this.damage, this);
-								} else {
-									DamageHandler.damageEntity(entity, this.damage, this);
+						if(!this.affectedEntities.contains(entity)) {
+							this.affectedEntities.add(entity);
+							if (this.damage != 0) {
+								if (entity instanceof LivingEntity) {
+									if (fstream.getAbility().getName().equalsIgnoreCase("AirSweep")) {
+										DamageHandler.damageEntity(entity, this.damage, this);
+									} else {
+										DamageHandler.damageEntity(entity, this.damage, this);
+									}
```

Comment:
Lines 206-210, can we get rid of the redundancy? I know you didn't write it, not blaming you. Just suggesting to add it to your changes.